ldes-server-pagination/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/pagination/batch/PageRelationProcessor.java

```diff
@@ -0,0 +1,82 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.pagination.batch;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.pagination.entities.Page;
+import be.vlaanderen.informatievlaanderen.ldes.server.pagination.entities.UnpagedMember;
+import be.vlaanderen.informatievlaanderen.ldes.server.pagination.repositories.PageRelationRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.pagination.repositories.PageRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.pagination.valueobjects.PageAssignment;
+import be.vlaanderen.informatievlaanderen.ldes.server.pagination.valueobjects.PartialUrl;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+@Component
+public class PageRelationProcessor implements ItemProcessor<List<UnpagedMember>, List<PageAssignment>> {
+	private final PageRepository pageRepository;
+	private final PageRelationRepository pageRelationRepository;
+
+	public PageRelationProcessor(PageRepository pageRepository, PageRelationRepository pageRelationRepository) {
+		this.pageRepository = pageRepository;
+		this.pageRelationRepository = pageRelationRepository;
+	}
+
+	@Override
+	public List<PageAssignment> process(List<UnpagedMember> items) {
+		if (items.isEmpty()) {
+			return List.of();
+		}
+
+		Page pageToFill = pageRepository.getOpenPage(items.getFirst().bucketId());
+
+		Queue<UnpagedMember> unpagedMembers = new ArrayDeque<>(items);
+		List<PageAssignment> pageAssignments = new ArrayList<>();
+
+		while (!unpagedMembers.isEmpty()) {
+			pageToFill = getPageWithSpace(pageToFill);
+			int membersAdded = fillPage(pageToFill, unpagedMembers.size());
+			for (int i = 0; i < membersAdded; i++) {
+				UnpagedMember member = unpagedMembers.poll();
+				assert member != null;
```

Comment:
Is it a good practice to use `assert` in source code? I have never seen it in our code as well, so maybe `Objects.requireNonNull` would be a better option here?